### PR TITLE
[MRG] By-pass set_annotations when plotting ICA sources 

### LINKED ISSUE #11762: Annotation displayed on ICA sources when meas_date is None and first_samp is not 0 are wrong
### Description of the problem

One more time-management related issue with annotations. MWE:

```
from datetime import datetime, timezone

import numpy as np
from mne import Annotations, create_info
from mne.io import RawArray
from mne.preprocessing import ICA


#%%
# date = datetime(1995, 3, 28, 0, 0, 0, 0, tzinfo=timezone.utc)

info = create_info(5, 1000, "eeg")
data = np.random.randn(5, 10000)
raw = RawArray(data, info)
assert raw.first_samp == 0
# raw.set_meas_date(date)
annotation = Annotations(onset=4, duration=1, description="bad_test")
raw.set_annotations(annotation)
# raw.plot(theme="light", scalings="auto")

#%%
raw.crop(2, 8)
# raw.plot(theme="light", scalings="auto")

#%%
ica = ICA(
    method="picard", fit_params=dict(ortho=False, extended=True), random_state=888
)
ica.fit(raw, picks="eeg", reject_by_annotation=True)
ica.plot_sources(raw, theme="light")
```

If there is no measurement date and if `raw.first_samp` is different from `0` (e.g. by cropping), the annotations displayed in `ica.plot_sources` are not correctly set. Issue is here: https://github.com/mne-tools/mne-python/blob/7def8369d7ca691d807fe916663168c2ee4c10e0/mne/viz/ica.py#L1304-L1305

Still thinking about what is a good fix.. time-management is a real headache :exploding_head:

### PR DESCRIPTION
**Pull request by mscheltienne**
No idea why my other branch is broken; but even on the website I can visualize it only once out of 5 tries.
Same as #11763 but working.

Fixes #11762

### CODE REVIEW

File: mne/viz/tests/test_ica.py [hunk 1]
```diff
@@ -328,6 +328,22 @@ def test_plot_ica_sources(raw_orig, browser_backend, monkeypatch):
...
+    raw.set_meas_date(None)
+    assert raw.first_samp != 0
+    raw.set_annotations(Annotations([0.2], [0.1], "Test"))
+    fig = ica.plot_sources(raw)
+    if browser_backend.name == "matplotlib":
+        assert len(fig.mne.ax_main.collections) == 1
+        assert len(fig.mne.ax_hscroll.collections) == 1
+    else:
+        assert len(fig.mne.regions) == 1
+        assert assert_allclose(fig.mne.regions[0].getRegion(), (0.2, 0.3))
```
Comment (larsoner) [inline mne/viz/tests/test_ica.py:344]: Don't forget this :)

Comment (mscheltienne) [reply mne/viz/tests/test_ica.py:344]: Damn I'm exhausted.. took me a while to notice the double `assert` with this change :see_no_evil:

**Comment by mscheltienne** [comment]
I don't usually merge my own PRs, but I'll consider this one reviewed by @larsoner in #11763.
I have someone on campus who would like to pull this change for his pipeline.

**Comment by larsoner** [comment]
Yes that's fine, thanks @mscheltienne !

**Pull request merged**